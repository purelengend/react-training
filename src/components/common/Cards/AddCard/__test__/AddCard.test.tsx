import { customRenderer, render } from '@src/test/test-utils';
import userEvent from '@testing-library/user-event';

import { AddCard } from '..';

describe('AddCard test cases', () => {
  it('should render correctly with onclick function', () => {
    const mockOnClick = vi.fn();

    const mockAddCard = customRenderer(<AddCard onClick={mockOnClick} />);

    expect(mockAddCard.toJSON()).toMatchSnapshot();
  });

  it('should invoke onclick function when being clicked', async () => {
    const mockOnClick = vi.fn();

    const { container } = render(<AddCard onClick={mockOnClick} />);

    const mockAddCard = container.querySelector('#add-card')!;

    await userEvent.click(mockAddCard);

    expect(mockOnClick).toBeCalled();
  });
});
