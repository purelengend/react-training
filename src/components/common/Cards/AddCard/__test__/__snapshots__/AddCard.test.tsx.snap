// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddCard test cases > should render correctly with onclick function 1`] = `
<div
  className="d-flex-center _addition-card_df6a58"
  id="add-card"
  onClick={[MockFunction spy]}
>
  <div
    className="d-flex-col _addition-wrapper_df6a58"
  >
    <img
      alt="Plus Icon"
      className="primary-icon"
      src="/src/assets/icons/plus-icon.svg"
    />
    <p
      className="_addition-content_df6a58"
    >
      Add new dish
    </p>
  </div>
</div>
`;
